### PR TITLE
MAT-4002 404 page for all measures component

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## MADiE PR

Jira Ticket: [MAT-4002](https://jira.cms.gov/browse/MAT-4002)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [x] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).
* [x] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/components/editMeasure/EditMeasure.tsx [hunk 1]
```diff
@@ -27,15 +28,27 @@ export default function EditMeasure() {
...
-        setLoading(false);
-      });
+      measureServiceApi
+        .fetchMeasure(id)
+        .then((value: Measure) => {
+          setMeasure(value);
+          setLoading(false);
+        })
+        .catch((err) => {
+          setNotFound(true);
```
[the remark this answers is not in the document]
Comment (mcmcphillips) [reply src/components/editMeasure/EditMeasure.tsx:43]: It looks like the measure service might need to be updated to return http status codes. 
The rest looks good to me. 
Should the measureservice status code changes be moved to another ticket?

[the remark this answers is not in the document]
Comment (adongare) [reply src/components/editMeasure/EditMeasure.tsx:43]: that will work

**Review by adongare — APPROVED** [review]

File: src/components/editMeasure/EditMeasure.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (sb-cecilialiu) [reply src/components/editMeasure/EditMeasure.tsx]: @adongare @mcmcphillips Thanks a lot for the comments, I have updated the code to check for status code 404. Please have a look again. Thanks!

File: src/components/editMeasure/EditMeasure.tsx [hunk 2]
```diff
@@ -27,15 +28,29 @@ export default function EditMeasure() {
...
+            setNotFound(true);
+          }
+        });
     }
   }, [measureServiceApi, id, measure]);
 
+  if (notFound) {
+    history.push("/404");
+  }
+
```
Comment (RohitKandimalla) [inline src/components/editMeasure/EditMeasure.tsx:53]: Need to be covered in unit test ?

File: src/components/notfound/NotFound.test.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import { Router } from "react-router-dom";
+import { createMemoryHistory } from "history";
+
+describe("NotFound component", () => {
+  it("should render NotFound component", async () => {
+    const { getByTestId, getByText } = render(
+      <MemoryRouter>
+        <NotFound />
+      </MemoryRouter>
+    );
```
Comment (RohitKandimalla) [inline src/components/notfound/NotFound.test.tsx:14]: I think we can render NotFound component without using MemoryRoute, you just need to mock useHistory with something like this.

Comment (RohitKandimalla) [reply src/components/notfound/NotFound.test.tsx:14]: Also the test cases doesn't have to async, I might be wrong but please double check.

Comment (sb-cecilialiu) [reply src/components/notfound/NotFound.test.tsx]: @RohitKandimalla Thanks so much for the comments. You are right those are not needed, so I removed them and just pushed in the code.

File: src/components/notfound/NotFound.test.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  });
+
+  it("should render home page after clicking Go Home", () => {
+    const history = createMemoryHistory();
+    const pushSpy = jest.spyOn(history, "push");
+    render(
+      <Router history={history}>
+        <NotFound />
+      </Router>
+    );
```
Comment (RohitKandimalla) [inline src/components/notfound/NotFound.test.tsx:24]: I think we could do that same here as well.

File: src/components/editMeasure/EditMeasure.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (sb-cecilialiu) [reply src/components/editMeasure/EditMeasure.tsx]: @RohitKandimalla @adongare @mcmcphillips So the check for error 404 is there, and I also removed the useState for notFound, to make the code cleaner, as Rohit suggested, thanks a lot!

File: src/components/editMeasure/EditMeasure.tsx (hunk 2 above)
Comment (sb-cecilialiu) [reply src/components/editMeasure/EditMeasure.tsx]: @RohitKandimalla The new test case is working, thanks so much for your kind help!!

File: src/components/notfound/NotFound.test.tsx (hunk 3 above)
Comment (sb-cecilialiu) [reply src/components/notfound/NotFound.test.tsx]: @RohitKandimalla The new code with mock push was checked in, thanks again Rohit!!!

File: src/components/notfound/NotFound.test.tsx (hunk 4 above)
Comment (sb-cecilialiu) [reply src/components/notfound/NotFound.test.tsx]: @RohitKandimalla Yes done. Thanks!

**Review by RohitKandimalla — APPROVED** [review]

**Review by adongare — DISMISSED** [review]

**Review by RohitKandimalla — APPROVED** [review]

**Pull request merged**